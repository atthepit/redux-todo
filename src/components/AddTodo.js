import React from 'react'
import { v4 } from 'uuid'

const AddTodo = ({ addTodo }) => {
  let textInput = undefined
  const onSubmit = (event) => {
    event.preventDefault()
    if (textInput == null)
      return

    const id = v4()
    const text = textInput.value
    addTodo(id, text)
    textInput.value = ``
  }

  return (
    <form onSubmit={ onSubmit }>
      <input
        ref={ (ref) => textInput = ref }
        type="text" />
      <button type="submit">{ `Add` }</button>
    </form>
  )
}

AddTodo.prototype.propTypes = {
  addTodo: React.PropTypes.func,
}

export default AddTodo
