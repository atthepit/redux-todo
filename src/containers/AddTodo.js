 import { connect } from 'react-redux'
 import { addTodo } from '../actions/todo'
 import AddTodo from '../components/AddTodo'

 const mapDispatchToProps = (dispatch) => ({
   addTodo: (id, text) => dispatch(addTodo(id, text)),
 })

 export default connect(
   null,
   mapDispatchToProps
 )(AddTodo)
