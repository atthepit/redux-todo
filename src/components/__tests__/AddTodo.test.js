import React from 'react'
import AddTodo from '../AddTodo'
import renderer from 'react-test-renderer'

describe(`Test AddTodo component`, () => {

  it(`should render correctly`, () => {
    const props = {
      addTodo: jest.fn(),
    }
    const tree = renderer.create(
      <AddTodo { ...props } />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
