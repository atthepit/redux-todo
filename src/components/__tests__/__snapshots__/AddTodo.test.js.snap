exports[`Test AddTodo component should render correctly 1`] = `
<form
  onSubmit={[Function]}>
  <input
    type="text" />
  <button
    type="submit">
    Add
  </button>
</form>
`;
